### PR TITLE
refactor: erc721

### PR DESCRIPTION
**Pull request by zhouhanseng**
#### This PR refactors erc-721 relative types, hooks:

1. Fetch the contract data and token data respectively: `useERC721ContractDetailed()` `useERC721TokenDetailed()`.
2. Redefine the ERC721 token type. split the property of contract and token.
3. Fetch data from opensea api first, if fails then fetch from chain, since the data of some NFT mint by opensea can't be retrieved from chain, and opensea provide the contract icon image and some other useful data.
4. Fix the dashboard 1.0 erc-721 Add Asset which is totally broken before, dedicate as a demo for dashboard 2.0.
5. Remove some unused erc-1155 logic, since the support for erc-1155 is even more unfinished than erc-721.
6. Rename `useTokenDetailed()` to `useFungibleTokenDetailed()`, `useTokenWatched()` to `useFungibleTokenWatched()`, since these two hooks only used for fungible token now. And It's better to use respective hook for NFT and FT.

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/database/types.ts [hunk 1]
```diff
@@ -13,29 +13,6 @@ export interface ERC20TokenRecord {
     symbol: string
 }
 
-export interface ERC721TokenRecord {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/database/types.ts:16]: Why remove it? We need ERC721 to store into DB.

File: packages/maskbook/src/plugins/Wallet/helpers.ts [hunk 2]
```diff
@@ -50,12 +44,12 @@ export function ERC20TokenArrayComparer(a: ERC20TokenRecord[], b: ERC20TokenReco
     return a.every((token, index) => ERC20TokenComparer(token, b[index]))
 }
 
-export function ERC721TokenComparer(a: ERC721TokenRecord | null, b: ERC721TokenRecord | null) {
+export function ERC721TokenComparer(a: ERC721TokenDetailed | null, b: ERC721TokenDetailed | null) {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/helpers.ts:47]: It should compare between DB records.

File: packages/maskbook/src/plugins/Wallet/services/assets.ts [hunk 3]
```diff
@@ -39,46 +39,29 @@ export async function getAssetsListNFT(
     provider: CollectibleProvider,
     page?: number,
     size?: number,
-) {
+): Promise<{ assets: ERC721TokenDetailed[]; hasNextPage: boolean }> {
     if (provider === CollectibleProvider.OPENSEAN) {
         const { assets } = await OpenSeaAPI.getAssetsList(address, { chainId, page, size })
         return {
             assets: assets
                 .filter((x) => ['ERC721', 'ERC1155'].includes(x.asset_contract.schema_name))
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/assets.ts:47]: It should filter ERC1155 tokens out if ERC1155 gave up.

File: packages/maskbook/src/plugins/Wallet/database/types.ts (hunk 1 above)
Comment (guanbinrui) [reply packages/maskbook/src/plugins/Wallet/database/types.ts:16]: It's moved.

Comment (zhouhanseng) [reply packages/maskbook/src/plugins/Wallet/database/types.ts:16]: Just replaced by ERC721TokenDetailed.

File: packages/maskbook/src/plugins/Wallet/services/assets.ts (hunk 3 above)
Comment (zhouhanseng) [reply packages/maskbook/src/plugins/Wallet/services/assets.ts:47]: oh done

**Review by guanbinrui — APPROVED** [review]

**Review by NikoManiac — APPROVED** [review]

**Pull request merged**